src/App.tsx

```diff
@@ -1,27 +1,55 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const memoizedGoodsList = useMemo(() => <GoodsList goods={goods} />, [goods]);
```

Comment:
You have already wrapped `GoodList` in `React.memo`.

We use `useMemo` to save our calculations for some data.
Check: https://legacy.reactjs.org/docs/hooks-reference.html#usememo